README.md

```diff
@@ -100,6 +100,101 @@ You are free to implement the genome classes however you want, and using whateer
+
+#### copy_te
+
+copy_te has a complexity of O(n*m).
+
+The complexity of this operation comes from calling insert_te. The other operations in the function should run in O(1), with the key-loop having a max complexity of O(p) if it runs through every key in the list.
+
+#### disable_te
+
+disable_te has a complexity of O(m).
+
+The key-loop has a maximal complexity of O(p), and changing the elements from 'A' to 'x' with a for-loop of length m runs in O(m). We will assume that m > p, because p depends on the order of the dictionary key list, and is therefore randomly sorted, while m is constant. The other operations in the function are simple, and are assumed to run in O(1) complexity.
+
+#### active_te
+
+active_te has a complexity of O(p)
+
+Getting the key list from the dictionary, and returning the result both run in O(1) complexity. We assume that we have to run through the key dict_list to get it into a list, in which case it wil have O(p) complexity.
+
+#### len
+
+len runs in O(1), because evaluating the length of a list has O(1) complexity.
+
+#### str
+
+str runs in O(n)
+
+str uses the join function, which has to run through each element in the genome to make it into a string. Therefore we expect the function ro run in O(n).
+
+### Doubly-Linked-List Genome
+
+#### init
+
+init has a complexity of O(n).
+
+Creating the genome by running through each element and appending takes n * O(1) = O(n) complexity. Creating hte ID and the TE-dictionary takes O(1).
+
+#### insert_te
+
+insert_te has a complecity of O(n)
+
+We run through each element in the key-list (O(p)), checking if the new te either overlaps, or pushes the existing te's. This runs in max O(p). If there is a overlap, we run the disable_te function, that run in O(n). Creating the new TE takes O(m). Finding the start, where we want to insert our new TE takes a maximal O(n) complexity, if start is at the end of the genome. Inserting the new TE takes O(1). All of these run-times can then be reduced to O(n).
+
+#### copy_te
+
+copy_te has a complexity of O(n).
+
+checking the keys has a maximum run-time of O(p), but our worst-case run-time must call the insert_te function. Therefore the run time of copy_te is defined by insert_te, which is defined by disable_te. Disable_te has a maximum complexity of O(n), and therefore, so does copy_te. 
```

Comment:
I agree. You could make it slightly more efficient, if offsets are small compared to the genome, if you didn't use `insert_te`.